modules/extensions-support/src/main/java/org/mule/module/extension/internal/model/property/UseConfigTypeModelProperty.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.model.property;
+
+
+import org.mule.extension.api.introspection.EnrichableModel;
+
+/**
+ * An immutable model property which specifies that the owning {@link EnrichableModel}
+ * requires a configuration of a given {@link #useConfigType}
+ *
+ * @since 4.0
+ */
+public final class UseConfigTypeModelProperty
```

Comment:
Rename to ConfigTypeModelProperty. The class name should reflect its purpose. The annotation should be free to change without impacting this class